src/client/meshgen/collector.cpp

```diff
@@ -95,10 +96,47 @@ PreMeshBuffer &MeshCollector::findBuffer(
 	if (numVertices > U16_MAX)
 		throw std::invalid_argument(
 				"Mesh can't contain more than 65536 vertices");
+
 	std::vector<PreMeshBuffer> &buffers = prebuffers[layernum];
+
+	// avoid scanning the entire list of buffers when filling with the same material
+	if (latest_buffers[layernum].size() > 0 && 
+			buffers[latest_buffers[layernum].back()].layer == layer && 
```

Comment:
Am I missing something or can you just replace that by `buffers.back()`?

Reply:
I've rewritten the code without using vectors of integers, much simpler now.